src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -73,58 +83,131 @@ public void storeMessage(Message message) {
+    Filter otherUserMessages =
+        CompositeFilterOperator.and(messagesSentByOtherUser, messagesReceivedByLoggedInUser);
 
     // All the messages between the two users
-    Filter directMessages = CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
+    return CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
+  }
 
-    // Messages are ordered from oldest to newest since messages between
-    // people may include implicit context/reference to previously sent messages,
-    // and this order makes the chronology of the conversation and context
-    // more intuitive and understandable in the messages list page. 
-    Query query =
-        new Query("Message")
-            .setFilter(directMessages)
-            .addSort("timestamp", SortDirection.ASCENDING);
+  public List<Message> getAllMessages() {
+    Query query = createPublicMessageQuery();
     PreparedQuery results = datastore.prepare(query);
+    return prepareQueryToMesssages(results);
+  }
 
-    for (Entity entity : results.asIterable()) {
-      String idString = entity.getKey().getName();
-      UUID id = UUID.fromString(idString);
-      String text = getStringProperty(entity, "text").orElse("");
-      long timestamp = (long) entity.getProperty("timestamp");
-      String sender = (String) entity.getProperty("user");
-      String receiver = (String) entity.getProperty("recipient");
-      Message message = new Message(id, sender, text, timestamp, receiver);
-      messages.add(message);
+  private List<Message> prepareQueryToMesssages(PreparedQuery query) {
+    List<Message> results = new ArrayList<>();
+    for (Entity entity : query.asIterable()) {
+      results.add(entityToMessage(entity));
     }
+    return results;
+  }
 
-    return messages;
+  private Message entityToMessage(Entity entity) {
+    String idString = entity.getKey().getName();
+    UUID id = UUID.fromString(idString);
+    String text = getStringProperty(entity, "text").orElse("");
+    long timestamp = (long) entity.getProperty("timestamp");
+    String sender = (String) entity.getProperty("user");
+    String receiver = (String) entity.getProperty("recipient");
+    return new Message(id, sender, text, timestamp, receiver);
+  }
+
+  private Query createPublicMessageQuery() {
+    return new Query("Message").addSort("timestamp", SortDirection.ASCENDING);
   }
 
-  /** Returns the total number of messages for all users. */
   public int getTotalMessageCount() {
     Query query = new Query("Message");
     PreparedQuery results = datastore.prepare(query);
     return results.countEntities(FetchOptions.Builder.withDefaults());
   }
 
+  /**
+   * This is to create an user form data to put in the datastore Going with option 2 highlighted in
+   * the Meeting Notes
+   *
+   * @throws EntityNotFoundException
+   */
+  public void storeUserTeaData(
+     Map<String, Long> incomingUserTeaData, String username, String date) {
+
+    Entity datastoreUser;
```

Comment:
The `datastoreUser` variable name is misleading; the variable does not represent "a user of the datastore" or "a user stored in the datastore", but rather the "tea data for a given user on a given date".  Since there are no other `Entity` variables in the local scope, I don't think it is necessary to disambiguate, and I would simply call this variable `entity`.